pkg/api/customization/monitor/handler_utils.go

```diff
@@ -237,7 +239,7 @@ func parseMetricParams(userContext *config.UserContext, nodeLister v3.NodeLister
 				}
 
 				for _, rc := range rcs.Items {
-					if rc.OwnerReferences != nil && strings.ToLower(rc.OwnerReferences[0].Kind) == workload.DeploymentType && rc.OwnerReferences[0].Name == name {
+					if len(rc.OwnerReferences) != 0 && strings.ToLower(rc.OwnerReferences[0].Kind) == workload.DeploymentType && rc.OwnerReferences[0].Name == name {
```

Comment:
If we are changing the logic here. Maybe we should use `worklaod.spec.selector` to selector pods. It would be better than using owner reference

Reply:
change to workload needs more update, we can change this in 2.2.3 together with the improvement for checking input params